### PR TITLE
[12.0] [MIG] [WIP] purchase_stock_picking_return_invoicing

### PR DESCRIPTION
**Pull request by antoniorubiopesol**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash all your migration commits and let the CIs in green.

**Comment by traviswaelbro** [comment]
@pedrobaeza Can this be merged?

**Comment by AaronHForgeFlow** [comment]
History is broken, no commits before migration and changes in other modules :confused:

File: purchase_stock_picking_return_invoicing/views/purchase_view.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                    <field name="invoice_refund_count"
+                           widget="statinfo"
+                           string="Refunds"
+                           attrs="{'invisible': [('state', 'in', ('draft', 'sent', 'to approve'))]}"
+                    />
+                </button>
+            </button>
+            <xpath expr="//field[@name='order_line']/tree/field[@name='qty_received']" position="after">
+                <field name="qty_returned"
+                       invisible="not context.get('show_purchase', False)"/>
```
Comment (traviswaelbro) [inline purchase_stock_picking_return_invoicing/views/purchase_view.xml:22]: This doesn't appear to work in Odoo 12. Instead, we can mirror the `attrs` used in core's Purchase Order form.

```suggestion
                       attrs="{'column_invisible': [('parent.state', 'not in', ('purchase', 'done'))]}"/>
```

File: purchase_stock_picking_return_invoicing/views/purchase_view.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            </xpath>
+            <xpath expr="//field[@name='order_line']/tree/field[@name='qty_invoiced']" position="after">
+                <field name="qty_refunded"
+                       invisible="not context.get('show_purchase', False)"/>
```
Comment (traviswaelbro) [inline purchase_stock_picking_return_invoicing/views/purchase_view.xml:26]: This doesn't appear to work in Odoo 12. Instead, we can mirror the `attrs` used in core's Purchase Order form.

```suggestion
                       attrs="{'column_invisible': [('parent.state', 'not in', ('purchase', 'done'))]}"/>
```

File: sale_timesheet_invoice_description/i18n/ar.po [hunk 3]
```diff
@@ -20,11 +20,14 @@ msgstr ""
 "&& n%100<=10 ? 3 : n%100>=11 && n%100<=99 ? 4 : 5;\n"
 
 #. module: sale_timesheet_invoice_description
+<<<<<<< HEAD
```
Comment (traviswaelbro) [inline sale_timesheet_invoice_description/i18n/ar.po:23]: Lots of leftover git conflict markers throughout the translation files.

```suggestion
```

**Review by traviswaelbro — CHANGES_REQUESTED** [review]

**Comment by JordiBForgeFlow** [comment]
@ageficent @hveficent @aheficent can you eview this one? I have an open PR in V11 with a change

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
Some comments:

Columns Returned Qty and Refunded Qty are not being displayed in the form view.
image

In Use Case 2 (according to Readme) after returning some quantities the Refunds button does not appear.
Peek 2019-07-09 09-21

There are some files from different modules that do not belong here (sale_timesheet_invoice_description)

**Comment by pedrobaeza** [comment]
Please superseed this as this doesn't contain commit history, which has been requested several times, and there seems to be no intention on fixing it.

**Comment by AaronHForgeFlow** [comment]
Apart of the issues already commented I do not get the purpose of this button (I don't know if this is introduced here but in runbot for OCA/account-invoicing for 12.0 this button is not there)
image

It lists all invoices for all suppliers, it seems to me this is not expected, and there is no explanation for the user.

**Comment by JordiBForgeFlow** [comment]
Looks like @antoniorubiopesol is not working on this PR anymore. I suggest that @ageficent @aheficent create a separate PR where the commit history is preserved.

**Comment by AdriaGForgeFlow** [comment]
> Looks like @antoniorubiopesol is not working on this PR anymore. I suggest that @ageficent @aheficent create a separate PR where the commit history is preserved.

On it :+1:

**Pull request closed without merging**